src/audio/copier/copier_gain.c

```diff
@@ -73,3 +73,79 @@ enum copier_gain_state copier_gain_eval_state(struct copier_gain_params *gain_pa
 
 	return state;
 }
+
+int copier_gain_dma_control(uint32_t node_id, const uint32_t *config_data,
+			    size_t config_size, enum sof_ipc_dai_type dai_type)
+{
+	union ipc4_connector_node_id node = (union ipc4_connector_node_id)node_id;
+	struct gain_dma_control_data *gain_data = NULL;
+	struct ipc_comp_dev *icd = NULL;
+	struct processing_module *mod = NULL;
+	struct copier_data *cd = NULL;
+	struct comp_dev *dev = NULL;
```

Comment:
The main problem here is grouping all declarations at the top of the function as opposed to declaring and initializing (and "const"ing when possible) in the same place like in any other programming language, including... C99. Avoiding C99 declarations in 2024 is a bit ridiculous, not sure why anyone would still be doing this. We've had  "accidental" C99 declarations in the SOF codebase since the dawn of time (#3995) and they never caused any issue with any toolchain, not even with our infamous (#7114) CAVS toolchain

Reply:
Seems reasonable